link.go

```diff
@@ -554,12 +563,117 @@ func (a *LinkStats64) unmarshalBinary(b []byte) error {
 	return nil
 }
 
+var (
+	// registeredDrivers is the global map of registered drivers
+	registeredDrivers = make(map[string]reflect.Type)
```

Comment:
Do we need reflect here?

I feel like this is unneeded and we can just use a `map[string]InterfaceDriver` here.
Is this to make a copy of the driver instance later in `getDriver`? Could we not solve that by derefencing before the asignement?

Reply:
I do not like this implementation, to be honest. I have tried dereferencing the interface but it did not work, 
`map[string]LinkDriver` holds the pointer for the registered instance and consequent calls to decrypt results in cumulative results. Fields set in prior calls stay the same. Maybe I am doing something wrong. I am open to suggestions.

Reply:
You are correct, we need a copy of the driver instance. Otherwise, the same instance is used in consequent calls to decrypt, and prior set fields stay unless they are not set again.
I am not fond of this implementation either. I have tried dereferencing and it does not work.
`t` as LinkDriver `newT := *t` throws `invalid operation: cannot indirect t (variable of type LinkDriver)`
I could not find any other way of doing this. I am open to suggestions.

Reply:
Hmm, this is a tricky one. To be honest, I am not quite sure. I thought that this would be possible without reflect, but now that I am testing stuff out I am not finding a proper solution...

All I could think of is adding a `Copy() LinkDriver` function to the interface definition. See https://go.dev/play/p/BcND_xYIpv6 for a basic explanation.

Reply:
Actually, `New() LinkDriver` would be a better name then `Copy`...

Reply:
This seems better than reflection. So I will go with it.